reactioncommerce/reaction
# Giving a user shopSettings permission does not allow them edit shop settings

**Resolves #2294** 

This happened because
- `shopSettings` permission was not amongst included permissions required for updates on Shops collection
- `Roles.userInRoles()` in email publication used to check if a user belongs to a particular group (with certain permissions) in a shop was returning false because it wasn't given a shopId required for checks. 

Test by
- Login as admin
- Create a new group called `customers` and Invite a user to that group
- Grant the `customers` group "Dashboard" , "Shop Settings" and "Email" permissions
- Log in as the user in step 2
- Try to edit shop settings by clicking on the shop icon on the toolbar menu
- Notice that a success toast message is shown and the set shop settings are changed
- Click on the email icon in the toolbar menu
- Notice that the email logs are visible but email settings are hidden
